src/dataset.jl

```diff
@@ -172,16 +172,132 @@ end
     write(dataset::AbstractDataset, filename::AbstractString; kwargs...)
 
 Writes the dataset to the designated filename.
+
+For the keyword arguments for vector datasets, consult the documentation of `writelayers` 
+and for raster datasets the docs for `writerasters`.
 """
 function write(
     dataset::AbstractDataset,
     filename::AbstractString;
     kwargs...,
 )::Nothing
+    
+    if nraster(dataset) > 0
+        destroy(unsafe_copy(dataset, filename = filename; kwargs...))
+    elseif nlayer(dataset) > 0
+        writelayers(dataset, filename; kwargs...)
+    elseif nraster(dataset) > 0 && nlayer(dataset) > 0
+        error("Writing datasets with raster and vector data is not supported.")
+    else
+        error("Could not determine dataset type (raster or vector).")
+    end
+    return nothing
+end
+
+"""
+    writerasters(dataset::AbstractDataset, filename::AbstractString; kwargs...)::Nothing
+
+Write a raster dataset to disk.
+
+### Parameters
+* `dataset`       the dataset to write
+
+### Keyword Arguments
+* `filename`      the filename, UTF-8 encoded.
+* `driver`        the driver to use
+* `strict`        ``true`` if the copy must be strictly equivalent, or more
+normally ``false`` if the copy may adapt as needed for the output format.
+* `options`       additional format dependent options controlling creation
+of the output file. `The APPEND_SUBDATASET=YES` option can be specified to
+avoid prior destruction of existing dataset.
+
+### Returns
+nothing
+"""
+function writerasters(dataset::AbstractDataset,
```

Comment:
If you want more time to think about whether you really want this, I can take it out again. Now it corresponds 1:1 to the old `write`.

Reply:
Ok, I'll remove it then and just improve the docs of `write` a bit.